src/dryrun.ts

```diff
@@ -81,7 +81,7 @@ export async function createDryrun({
       if (t.txn.appForeignAssets) assets.push(...t.txn.appForeignAssets);
 
       // Create application,
-      if (t.txn.appIndex === 0) {
+      if (t.txn.appIndex === undefined) {
```

Comment:
True... I I know `algosdk.createDryrun` will have `appIndex` as `undefined` for creation transactions. If someone is sourcing the dry run request from elsewhere, however, it could be 0. Not sure how `goal` or other SDKs behave.

Reply:
Thanks for spotting this - I agree that we might want to err on caution and include both checks for:

```suggestion
      if (t.txn.appIndex === undefined || t.txn.appIndex === 0) {
```